### PR TITLE
Create Marker Plots

### PR DESCRIPTION
**Pull request by rnmitchell**
This PR will create a single PDF file containing the reads vs. CE allele for each locus.

### CODE REVIEW

File: lusSTR/wrappers/convert.py [hunk 1]
```diff
@@ -170,6 +175,49 @@ def sort_table(table):
...
+            pdf.savefig()
+
+
+def make_plot(df, id, sex, sameyaxis):
+    sample_df = df[df["SampleID"] == id]
+    sample_id = f"{id}_sexchr" if sex else id
+    max_reads = max(sample_df["Reads"])
+    n = 100 if max_reads > 1000 else 10
+    max_yvalue = int(math.ceil(max_reads / n)) * n
+    increase_value = int(math.ceil((max_yvalue / 5)) / n) * n
```
Comment (rnmitchell) [inline lusSTR/wrappers/convert.py:196]: This is to make nice round values (10, 20, 100, 200, etc.) when determining the max y value of the dataset and what the incremental increase should be for the plots. The first line determines whether to round to 10s or 100s depending on the highest # of reads.

**Comment by rnmitchell** [comment]
RebJ had requested the two separate pages: one showing the individual marker plots with its own y-axis scale (better for examining individual marker plots) and one showing the individual marker plots with the same y-axis scale (better for getting an idea of the profile as a whole).

**Comment by rnmitchell** [comment]
This is ready for review @standage. I'm attaching two examples of the plots, one for the autosomal STRs and one for the sex chr STRs since you aren't able to see what they look like.
[Uploading lusstr_output_Positive_Control_marker_plots.pdf…]()
[Uploading lusstr_output_Positive_Control_sexchr_marker_plots.pdf…]()

File: README.md [hunk 2]
```diff
@@ -165,7 +165,12 @@ Further, a second table (labeled as ```*_flanks.txt```) containing information r
 *  3' Flanking Sequence Bracketed Notation
 *  Potential Issues (such as: Possible indel or partial sequence)
 
-The ```Potential Issues``` column in this report is to draw attention to potential problem sequences (due to perhaps an indel or partial sequence) and requires the attention of the user to further evaluate the sequence for it's authenticity.
+The ```Potential Issues``` column in this report is to draw attention to potential problem sequences (due to perhaps an indel or partial sequence) and requires the attention of the user to further evaluate the sequence for it's authenticity.
```
Comment (standage) [inline README.md:168]: Typo: "for *its* authenticity".

File: lusSTR/wrappers/convert.py [hunk 3]
```diff
@@ -170,6 +175,48 @@ def sort_table(table):
...
+        with PdfPages(f"MarkerPlots/{output_name}_{sample_id}_marker_plots.pdf") as pdf:
+            make_plot(df, id, sex, False)
+            pdf.savefig()
+            make_plot(df, id, sex, True)
+            pdf.savefig()
+
+
+def make_plot(df, id, sex, sameyaxis):
+    sample_df = df[df["SampleID"] == id]
+    sample_id = f"{id}_sexchr" if sex else id
```
Comment (standage) [inline lusSTR/wrappers/convert.py:192]: Suggestion for a marginal improvement in readability: make the `sameyaxis` argument a keyword argument by giving it a default value.

File: lusSTR/wrappers/convert.py [hunk 4]
```diff
@@ -170,6 +175,48 @@ def sort_table(table):
...
+def marker_plots(df, output_name, sex):
+    Path("MarkerPlots").mkdir(parents=True, exist_ok=True)
+    df["CE_Allele"] = df["CE_Allele"].astype(float)
+    for id in df["SampleID"].unique():
+        sample_id = f"{id}_sexchr" if sex else id
+        with PdfPages(f"MarkerPlots/{output_name}_{sample_id}_marker_plots.pdf") as pdf:
+            make_plot(df, id, sex, False)
+            pdf.savefig()
+            make_plot(df, id, sex, True)
+            pdf.savefig()
```
Comment (standage) [inline lusSTR/wrappers/convert.py:187]: That way, when someone is messing with the code 18 months from now and they're trying to remember what that dang 4th argument to the `make_plot` function is, passing the argument as a key/value pair will provide a great contextual reminder.

```python
make_plot(df, id, sex, sameyaxis=False)
```

File: lusSTR/wrappers/convert.py [hunk 5]
```diff
@@ -202,6 +250,7 @@ def main(input, out, kit, uas, sex, nocombine):
             autosomal_final_table.to_csv(out, sep="\t", index=False)
     else:
         autosomal_final_table.to_csv(out, sep="\t", index=False)
+    marker_plots(autosomal_final_table, output_name, False)
```
Comment (standage) [inline lusSTR/wrappers/convert.py:253]: Similar suggestion with the `sex` argument here.

```python
marker_plots(autosomal_final_table, output_name, sex_chr=False)
```

**Review by standage — COMMENTED** [review]
Looks good, just a couple of suggestions for your consideration.

**Comment by rnmitchell** [comment]
Updated!

**Review by standage — APPROVED** [review]

**Pull request merged**